src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java

```diff
@@ -192,6 +200,51 @@ private void proceed() {
 		});
 	}
 	
+	public void initializeShowHideProfile(boolean isOwner) {
+		if (isOwner) {
+			boolean isProfileVisible = true;
+			try {
+				String cookieValue = cookies.getCookie(USER_PROFILE_VISIBLE_STATE_KEY);
+				if (cookieValue != null && !cookieValue.isEmpty()) {
+					isProfileVisible = Boolean.valueOf(cookieValue);	
+				}
+			} catch (Exception e) {
+				//if there are any problems getting the profile visibility state, ignore and use default (show)
+			}
+			setIsProfileVisible(isProfileVisible, false);
+		} else {
+			//not the owner
+			//show the profile, and hide the profile button
+			setIsProfileVisible(true, false);
+			view.setHideProfileButtonVisible(false);
+		}
+	}
+	
+	@Override
+	public void hideProfileButtonClicked() {
+		setIsProfileVisible(false, true);
+	}
+	
+	@Override
+	public void showProfileButtonClicked() {
+		setIsProfileVisible(true, true);
+	}
+	
+	private void setIsProfileVisible(boolean isVisible, boolean isSetCookie) {
+		if (isVisible){
+			view.showProfile();
+		} else {
+			view.hideProfile();
+		}
+		view.setShowProfileButtonVisible(!isVisible);
+		view.setHideProfileButtonVisible(isVisible);
+		if (isSetCookie) {

```

Comment:
Setting the cookie could be its own method called only when the show/hide buttons are pressed.